src/DataSet/ObjectDataSet.php

```diff
@@ -77,9 +86,6 @@ private function parseObject(): void
         $this->rules = $objectHasRules ? $this->object->getRules() : [];
 
         $this->dataSetProvided = $this->object instanceof DataSetInterface;
-        if ($this->dataSetProvided) {
-            return;
-        }
```

Comment:
If object instance of `DataSetInterface` then we don't need parse object.

Reply:
No, we do need to parse the object at least for the rules. Otherwise some tests will fail.

Empty rules, no errors while rules are declared via attributes:

https://github.com/yiisoft/validator/blob/616987bfef8820a5292086064308b918af88c51d/tests/ValidatorTest.php#L291-L299
https://github.com/yiisoft/validator/blob/616987bfef8820a5292086064308b918af88c51d/tests/DataSet/ObjectDataSetTest.php#L79

The first test is new, and the second is existing but assert was wrong.

Reply:
If object provide data via implementation of `DataSetInterface` then we don't can parse rules from properties, because data is not properties.